File: android/api-tests/src/test/java/com/revenuecat/apitests/java/ExperimentalCommonApiTests.java
package com.revenuecat.apitests.java;

import com.revenuecat.purchases.hybridcommon.ExperimentalCommonKt;

import java.util.Map;

@SuppressWarnings("unused")
class ExperimentalCommonApiTests {
    private void checkTrackAdDisplayed(Map<String, Object> adData) {
        ExperimentalCommonKt.trackAdDisplayed(adData);
    }

    private void checkTrackAdOpened(Map<String, Object> adData) {
        ExperimentalCommonKt.trackAdOpened(adData);
    }

    private void checkTrackAdRevenue(Map<String, Object> adData) {
        ExperimentalCommonKt.trackAdRevenue(adData);
    }

    private void checkTrackAdLoaded(Map<String, Object> adData) {
        ExperimentalCommonKt.trackAdLoaded(adData);
    }

    private void checkTrackAdFailedToLoad(Map<String, Object> adData) {
        ExperimentalCommonKt.trackAdFailedToLoad(adData);
    }
}

File: android/api-tests/src/test/java/com/revenuecat/apitests/kotlin/ExperimentalCommonApiTests.kt
package com.revenuecat.apitests.kotlin

import com.revenuecat.purchases.hybridcommon.ExperimentalPreviewHybridCommonAPI
import com.revenuecat.purchases.hybridcommon.trackAdDisplayed
import com.revenuecat.purchases.hybridcommon.trackAdFailedToLoad
import com.revenuecat.purchases.hybridcommon.trackAdLoaded
import com.revenuecat.purchases.hybridcommon.trackAdOpened
import com.revenuecat.purchases.hybridcommon.trackAdRevenue

@Suppress("unused")
@OptIn(ExperimentalPreviewHybridCommonAPI::class)
private class ExperimentalCommonApiTests {
    fun checkTrackAdDisplayed(adData: Map<String, Any?>) {
        trackAdDisplayed(adData)
    }

    fun checkTrackAdOpened(adData: Map<String, Any?>) {
        trackAdOpened(adData)
    }

    fun checkTrackAdRevenue(adData: Map<String, Any?>) {
        trackAdRevenue(adData)
    }

    fun checkTrackAdLoaded(adData: Map<String, Any?>) {
        trackAdLoaded(adData)
    }

    fun checkTrackAdFailedToLoad(adData: Map<String, Any?>) {
        trackAdFailedToLoad(adData)
    }
}
